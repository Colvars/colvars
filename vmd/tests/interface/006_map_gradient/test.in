colvar {
    name numatoms
    outputAppliedForce yes
    mapTotal {
        name numatoms
        mapID 0
        atoms {
            atomNumbers 1
        }
    }
}
