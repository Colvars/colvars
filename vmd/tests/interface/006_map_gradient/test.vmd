# -*- tcl -*-

set sigma 2.5
set res 0.2

set f [open "zero.pdb" "w"]
puts $f "ATOM      1  CA  ALA B   1       0.000   0.000   0.000  1.00  1.00      CA    
END
"
close $f

set f [open "oneatom.pdb" "w"]
puts $f "ATOM      1  CA  ALA B   1       1.450   0.000   0.000  1.00  1.00      CA    
END
"
close $f

mol new zero.pdb
set sel [atomselect top "(all)"]
${sel} set radius [expr ${sigma} * 2.0]
volmap \
    density \
    ${sel} \
    -minmax { { -5.0 -5.0 -5.0} {  5.0  5.0  5.0} } \
    -res ${res} \
    -allframes \
    -combine avg \
    -checkpoint 0 \
    -o gaussian-fine.density.dx
${sel} delete
mol delete top

mol new oneatom.pdb waitfor all
mol addfile gaussian-fine.density.dx type dx waitfor all

set pi [expr acos(-1.0)]
# set cellvol [voltool info -mol 1 -vol 0 cellvolume]
# set nx [voltool info -mol 1 -vol 0 xsize]
# set ny [voltool info -mol 1 -vol 0 ysize]
# set nz [voltool info -mol 1 -vol 0 zsize]
set norm [expr 1.0/(sqrt(8.0*${pi}*${pi}*${pi})*${sigma}*${sigma}*${sigma})]

#set norm [lindex [voltool info -mol 1 -vol 0 minmax] 1]
puts $norm

set sel [atomselect top "(all)"]

set xcoord [lindex [$sel get x] 0]

cv molid top
cv configfile test.in

cv colvar numatoms set collect_gradient 1
cv update

puts ""
puts [format "colvars: map colvar = %12.9f" [cv colvar numatoms value]]
puts [format "colvars: interpvol0 = %12.9f" [${sel} get interpvol0]]
set gaussian_value [expr ${norm}*exp(-0.5*(${xcoord}*${xcoord})/(${sigma}*${sigma}))]; list
puts [format "colvars:     theory = %12.9f" ${gaussian_value}]
puts ""

puts [format "colvars: grad/theory = %12.9f %12.9f %12.9f" [expr ${gaussian_value} * (-${xcoord})/(${sigma}*${sigma})] 0.0 0.0]

puts [format "colvars: grad/colvar = %12.9f %12.9f %12.9f" {*}[lindex [cv colvar numatoms getgradients] 0]]

puts ""

file delete -force gaussian-fine.density.dx
file delete -force zero.pdb
file delete -force oneatom.pdb

quit
