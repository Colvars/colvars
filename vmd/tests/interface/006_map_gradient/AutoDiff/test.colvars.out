colvars: ----------------------------------------------------------------------
colvars: Initializing the collective variables module, version "2020-09-25".
colvars: Please cite Fiorin et al, Mol Phys 2013:
colvars:  https://dx.doi.org/10.1080/00268976.2013.813594
colvars: in any publication based on this calculation.
colvars: This version was built without the C++11 standard: some features are disabled.
colvars: Please see the following link for details:
colvars: https://colvars.github.io/README-c++11.html
colvars: Using VMD interface, version "2020-09-25".
colvars: Redefining the Tcl "cv" command to the new script interface.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration from file "test.in":
colvars: # units = "" [default]
colvars: # smp = on [default]
colvars: # colvarsTrajFrequency = 0 [default]
colvars: # colvarsRestartFrequency = 0 [default]
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = off [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "numatoms"
colvars:   Initializing a new "mapTotal" component.
colvars:     # name = "numatoms"
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:     # mapName = "" [default]
colvars:     # mapID = 0
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 1 atoms requested: total mass = 12.011, total charge = 0.
colvars:     # atomWeights =  [default]
colvars:   All components initialized.
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 1 [default]
colvars:   # lowerBoundary = 0 [default]
colvars:   # upperBoundary = 0 [default]
colvars:   # hardLowerBoundary = off [default]
colvars:   # hardUpperBoundary = off [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = on
colvars:   # subtractAppliedForce = off [default]
colvars:   # runAve = off [default]
colvars:   # corrFunc = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 1 in total.
colvars: No collective variables biases were defined.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: Re-initialized atom group for variable "numatoms":0/0. 1 atoms: total mass = 12.011, total charge = 0.
colvars: map colvar =  0.003427587
colvars: interpvol0 =  0.003427587
colvars:     theory =  0.003434487
colvars: grad/theory = -0.000796801  0.000000000  0.000000000
colvars: grad/colvar = -0.000792025  0.000000000  0.000000000
